app/models/audience_segment.rb

```diff
@@ -0,0 +1,64 @@
+class AudienceSegment < ApplicationRecord
+  # enum does not like names that start with "not_"
+  enum type_of: {
+    manual: 0, # never matches anyone, used in test factory
+    trusted: 1,
+    posted: 2,
+    no_posts_yet: 3,
+    dark_theme: 4,
+    light_theme: 5,
+    no_experience: 6,
+    experience1: 7,
+    experience2: 8,
+    experience3: 9,
+    experience4: 10,
+    experience5: 11
+  }
+
+  has_many :segmented_users, dependent: :destroy
+  has_many :users, through: :segmented_users
+
+  after_validation :run_query
+
+  QUERIES = {
```

Comment:
This is the one area I am still having internal conflict about.

If we roll with this where we have a single audience segment per billboard, then we could still build out a potential for targeting multiple segments in the future, perhaps via a DSL.

I think eventually we probably do want the multiple, and maybe it is easier to build that up front, but that also means we're immediately committing to a join table or serialized relationship right off the bat.

It's hard to know, but since audience segment should likely be an AND relationship, where with tags we have an OR relationship, I feel like we want to take slow steps here instead of creating too big a manual as far as how the different functionalities work.

I still have consternation about this, but I think keeping it a simple belongs_to relationship for now and dealing with the tradeoffs later when we get a better sense of what matters most. Either way, we'll get a lot of mileage out of the "manual" segment in the future.